### PR TITLE
Extend IContentCategories behavior with a new field new content categories

### PR DESCRIPTION
**Pull request by maethu**
- Add new content categories should work the same way in AT and DX.
- This PR adds a new field called `new content categories.`
- Implements it own behavior storage (new content categories gets stored in content categories).
- Should have no impact at all (indexing, etc.)
- Protect the new add content categories field with a permission

The new permission is not managed by lawgiver on purpose.
Because it should act like the RolesAllowedToAddNewKeyworts
global configuration on AT KeywordWidgets.

### CODE REVIEW

**Comment by maethu** [comment]
@jone can you take a look?

File: ftw/contentpage/behaviors/configure.zcml [hunk 1]
```diff
@@ -12,7 +12,7 @@
         title="ftw.contentpage content categories behavior"
         description="Extends a content by the content_categories field"
         provides="ftw.contentpage.behaviors.content_categories.IContentCategories"
-        factory="plone.behavior.AnnotationStorage"
+        factory=".content_categories.ContentCategoriesStorage"
```
Comment (jone) [inline ftw/contentpage/behaviors/configure.zcml]: Is the data still at the same place? Is a migration required?

Comment (maethu) [inline ftw/contentpage/behaviors/configure.zcml]: @jone Nope - I use the annotation blabla impl (https://github.com/4teamwork/ftw.contentpage/pull/211/files#diff-a5b966409543b61ce8841f4c2305986bR42). The dexterity annotation factory is using the same.

Comment (maethu) [inline ftw/contentpage/behaviors/configure.zcml]: @jone but I don't know why the tests are failing on the CI. hold on.

**Comment by maethu** [comment]
@jone I added a write_permission for the new add categories field. 

And managed the plone 4.1 and 4.2 test to work again. 
I hacked, do you habe a better idea (solution)?

File: ftw/contentpage/behaviors/configure.zcml [hunk 2]
```diff
@@ -4,16 +4,19 @@
...
-    <include package="plone.directives.form" />
 
-    <plone:behavior
-        title="ftw.contentpage content categories behavior"
-        description="Extends a content by the content_categories field"
-        provides="ftw.contentpage.behaviors.content_categories.IContentCategories"
-        factory="plone.behavior.AnnotationStorage"
-        marker="ftw.contentpage.interfaces.ICategorizable"
-        />
+    <configure zcml:condition="have plone-43">
```
Comment (jone) [inline ftw/contentpage/behaviors/configure.zcml]: you wanted to move this up to the package root, right?

Comment (maethu) [inline ftw/contentpage/behaviors/configure.zcml]: @jone yep, but it does not solve the issues with the indexer.py and viewlets.py... so imho it doesn't matters.

**Pull request merged**